src/datastore/common.ts

```diff
@@ -384,67 +386,75 @@ export interface DbPox2DelegateStackStxEvent extends DbPox2BaseEventData {
   };
 }
 
-export interface DbPox2DelegateStackIncreaseEvent extends DbPox2BaseEventData {
-  name: Pox2EventName.DelegateStackIncrease;
+export interface DbPoxSyntheticDelegateStackIncreaseEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.DelegateStackIncrease;
   data: {
     increase_by: bigint;
     total_locked: bigint;
     delegator: string;
   };
 }
 
-export interface DbPox2DelegateStackExtendEvent extends DbPox2BaseEventData {
-  name: Pox2EventName.DelegateStackExtend;
+export interface DbPoxSyntheticDelegateStackExtendEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.DelegateStackExtend;
   data: {
     unlock_burn_height: bigint;
     extend_count: bigint;
     delegator: string;
   };
 }
 
-export interface DbPox2StackAggregationCommitEvent extends DbPox2BaseEventData {
-  name: Pox2EventName.StackAggregationCommit;
+export interface DbPoxSyntheticStackAggregationCommitEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.StackAggregationCommit;
   data: {
     reward_cycle: bigint;
     amount_ustx: bigint;
   };
 }
 
-export interface DbPox2StackAggregationCommitIndexedEvent extends DbPox2BaseEventData {
-  name: Pox2EventName.StackAggregationCommitIndexed;
+export interface DbPoxSyntheticStackAggregationCommitIndexedEvent
+  extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.StackAggregationCommitIndexed;
   data: {
     reward_cycle: bigint;
     amount_ustx: bigint;
   };
 }
 
-export interface DbPox2StackAggregationIncreaseEvent extends DbPox2BaseEventData {
-  name: Pox2EventName.StackAggregationIncrease;
+export interface DbPoxSyntheticStackAggregationIncreaseEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.StackAggregationIncrease;
   data: {
     reward_cycle: bigint;
     amount_ustx: bigint;
   };
 }
 
-export type DbPox2EventData =
-  | DbPox2HandleUnlockEvent
-  | DbPox2StackStxEvent
-  | DbPox2StackIncreaseEvent
-  | DbPox2StackExtendEvent
-  | DbPox2DelegateStxEvent
-  | DbPox2DelegateStackStxEvent
-  | DbPox2DelegateStackIncreaseEvent
-  | DbPox2DelegateStackExtendEvent
-  | DbPox2StackAggregationCommitEvent
-  | DbPox2StackAggregationCommitIndexedEvent
-  | DbPox2StackAggregationIncreaseEvent;
+export interface DbPoxSyntheticRevokeDelegateStxEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.RevokeDelegateStx;
+  data: {
+    // TODO: determine what data is available for this event type
+    amount_ustx: bigint;
+    delegate_to: string;
```

Comment:
I'm assuming only the sender address is available (given pox clar code), but not sure how much more the node will provide.

Reply:
The WIP PR https://github.com/stacks-network/stacks-core/pull/4157 is likely to add more data. There will probably be a follow-up API PR once that is ready